raven/processes/wps_nalcms_zonal_stats.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import json
+import tempfile
+
+from pywps import LiteralInput, ComplexInput
+from pywps import ComplexOutput
+from pywps import Process, FORMATS
+from pywps.app.Common import Metadata
+from rasterstats import zonal_stats
+
+from raven.utils import archive_sniffer, crs_sniffer, generic_vector_reproject, single_file_check
+
+LOGGER = logging.getLogger("PYWPS")
+
+NALCMS_CATEGORIES = {
+    1: 'Forest',
+    2: 'Forest',
+    3: 'Forest',
+    4: 'Forest',
+    5: 'Forest',
+    6: 'Forest',
+    7: 'Shrubs',
+    8: 'Shrubs',
+    9: 'Grass',
+    10: 'Grass',
+    11: 'Shrubs',
+    12: 'Grass',
+    13: 'Grass',
+    14: 'Wetland',
+    15: 'Crops',
+    16: 'Grass',
+    17: 'Urban',
+    18: 'Water',
+    19: 'SnowIce',
+}
+
+
+class NALCMSZonalStatisticsProcess(Process):
+    """Given files containing vector data and raster data, perform zonal statistics of the overlapping regions"""
+
+    def __init__(self):
+        inputs = [
+            ComplexInput('shape', 'Vector Shape',
+                         abstract='An ESRI Shapefile, GML, JSON, GeoJSON, or single layer GeoPackage.'
+                                  ' The ESRI Shapefile must be zipped and contain the .shp, .shx, and .dbf.'
+                                  ' The shape and raster should have a matching CRS.',
+                         min_occurs=1, max_occurs=1,
+                         supported_formats=[FORMATS.GEOJSON, FORMATS.GML, FORMATS.JSON, FORMATS.SHP]),
+            ComplexInput('raster', 'Gridded Land Use raster data set',
```

Comment:
Once the DEM is on GeoServer, this will not be necessary correct ?

Reply:
Correct. It would be interesting to provide different years [2000, 2010, 2020(?)] for the NALCMS data set but they would all be served via WCS provided by the PAVICS instance.